pkg/network/cluster_config.go

```diff
@@ -117,9 +123,128 @@ func validateClusterConfig(clusterConfig *configv1.Network, infraRes *bootstrap.
 	}
 
 	if _, ok := clusterConfig.Annotations[names.NetworkTypeMigrationAnnotation]; ok {
-		// HostedControlPlane is not nil if in a HyperShift env
-		if infraRes.HostedControlPlane != nil {
-			return errors.Errorf("network type live migration is not supported on HyperShift clusters")
+		return validateLiveMigration(clusterConfig, infraRes, client)
+	}
+
+	return nil
+}
+
+// validateCIDROverlap validates whether any of the ClusterNetwork and ServiceNetwork CIDRs overlap with
+// the internal CIDRs used by OVNKubernetes
+func validateOVNKubernetesCIDROverlap(clusterCofnig *configv1.Network, operConfig *operv1.Network) error {
+	// Verify whether the available subnets conflict between CNIs
+	type subnet struct {
+		name string
+		cidr string
+	}
+	var subnets []subnet
+	for _, cn := range clusterCofnig.Spec.ClusterNetwork {
+		subnets = append(subnets, subnet{
+			name: "clusterNetwork",
+			cidr: cn.CIDR,
+		})
+	}
+	for _, svcNet := range clusterCofnig.Spec.ServiceNetwork {
+		subnets = append(subnets, subnet{
+			name: "serviceNetwork",
+			cidr: svcNet,
+		})
+	}
+
+	v4InternalSubnet, v6InternalSubnet := GetInternalSubnets(operConfig.Spec.DefaultNetwork.OVNKubernetesConfig)
```

Comment:
Do we need to check the IPv6 subnet? We don't support IPv6 in live migration.

Reply:
I think it doesn't hurt to have if here, there is a chance that a creative user sets the IPv6 CIDRs to overlapping values before kicking of the migration right?

Reply:
We don't support SDN to dual-stack OVN migration. We may reject it here.

Reply:
The user can change the cluster to dual-stack after migrating from SDN, there is no way to make the cluster dual-stack during the migration.
I don't think that there is any harm in having the ipv6 checks in place here although I agree that this is an unlikely scenario.